lib/helpers/handleFileTracking.js

```diff
@@ -0,0 +1,80 @@
+const { join } = require("path");
+const {
+  existsSync,
+  readdirSync,
+  readFileSync,
+  writeFileSync,
+  removeSync,
+} = require("fs-extra");
+const findCacheDir = require("find-cache-dir");
+const { NETLIFY_PUBLISH_PATH, NETLIFY_FUNCTIONS_PATH } = require("../config");
+
+const TRACKING_FILE_SEPARATOR = "---";
+
+// Clean configured publish and functions folders and track next-on-netlify files
+// for future cleans
+const handleFileTracking = ({ functionsPath, publishPath }) => {
+  const isConfiguredFunctionsDir = functionsPath !== NETLIFY_FUNCTIONS_PATH;
+  const isConfiguredPublishDir = publishPath !== NETLIFY_PUBLISH_PATH;
+
+  const cacheDir = findCacheDir({ name: "next-on-netlify", create: true });
+  const trackingFilePath = join(cacheDir, ".nonfiletracking");
+
+  if (existsSync(trackingFilePath)) {
+    const trackingFile = readFileSync(trackingFilePath, "utf8");
+    const [trackedFunctions, trackedPublish] = trackingFile.split(
+      TRACKING_FILE_SEPARATOR
+    );
+
+    const cleanConfiguredFiles = (trackedFiles) => {
+      trackedFiles.forEach((file) => {
+        const filePath = join(publishPath, file);
+        if (file !== "" && existsSync(filePath)) {
+          removeSync(filePath);
+        }
+      });
+    };
+
+    if (isConfiguredPublishDir) {
+      cleanConfiguredDir(trackedPublish.split("\n"));
```

Comment:
..................................how did the tests pass wtf

Reply:
oh. .nonfiletracking will never exist at first in the test environment because its always a brand new build